pkg/crc/api/handlers.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	"github.com/code-ready/crc/pkg/crc/version"
+)
+
+func statusHandler(_ ArgsType) string {
+	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
+	clusterStatus, err := machine.Status(statusConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(clusterStatus)
+}
+
+func stopHandler(_ ArgsType) string {
+	stopConfig := machine.StopConfig{
+		Name:  constants.DefaultName,
+		Debug: true,
+	}
+	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(commandResult)
+}
+
+func startHandler(_ ArgsType) string {
+	startConfig := machine.StartConfig{
+		Name:          constants.DefaultName,
+		BundlePath:    crcConfig.GetString(config.Bundle.Name),
+		VMDriver:      crcConfig.GetString(config.VMDriver.Name),
+		Memory:        crcConfig.GetInt(config.Memory.Name),
+		CPUs:          crcConfig.GetInt(config.CPUs.Name),
+		NameServer:    crcConfig.GetString(config.NameServer.Name),
+		GetPullSecret: getPullSecretFileContent,
+		Debug:         true,
+	}
+	status, err := machine.Start(startConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	return encodeStructToJson(status)
+}
+
+func versionHandler(_ ArgsType) string {
+	v := &machine.VersionResult{
+		CrcVersion:       fmt.Sprintf("%s+%s", version.GetCRCVersion(), version.GetCommitSha()),
+		OpenshiftVersion: version.GetBundleVersion(),
```

Comment:
For what it's worth, this `GetBundleVersion()` is the bundle version the binary was built for (and the one which is embedded). At the moment, we artificially prevent from running a different bundle version, but the day this changes, this code will need to be updated.

Reply:
Right, this will need to be updated, but will depend on whether we change implementation in the same function, change the function signature, or make a different function to get the openshift version.